### PR TITLE
Accept response validation error handler

### PR DESCRIPTION
**Pull request by nampas**
Here's a stab at the feature outlined here: https://github.com/cdimascio/express-openapi-validator/issues/454

This PR adds a new `validateResponses.onError` option. There's no strong contract with the function, but express-openapi-validator will pass the `err` object, as well as the response `data`. Usage might look something like this

```
const handler = (err, data) => {
   console.log('OpenAPI response validation error', err);
};

OpenApiValidator.middleware({
   apiSpec: './spec/openapi.json',
   validateRequests: true,
   validateResponses: { onError: handler }
});
```

### CODE REVIEW

File: src/middlewares/openapi.response.validator.ts [hunk 1]
```diff
@@ -60,13 +69,23 @@ export class ResponseValidator {
...
+            validators,
+            body,
+            statusCode,
+            path,
+            accepts, // return 406 if not acceptable
+          });
+        } catch (err) {
+          // If a custom error handler was provided, we call that
+          if (err instanceof InternalServerError && this.eovOptions.onError) {
+            this.eovOptions.onError(err, req, res, next, body)
```
Comment (nampas) [inline src/middlewares/openapi.response.validator.ts:83]: I think maybe we shouldn't pass `next` here, as the wrapper in `mung` is responsible for invoking that on a successful validation.

Comment (cdimascio) [reply src/middlewares/openapi.response.validator.ts:83]: can you describe this a bit more

Comment (cdimascio) [reply src/middlewares/openapi.response.validator.ts:83]: i also wonder if `err` alone is sufficient. is req, res, body needed?

Comment (nampas) [reply src/middlewares/openapi.response.validator.ts:83]: I think you're right. I don't seen a need for req, res and next. I do think body could be useful though. For example, users may want to extract something from it to log, (e.g. a resource id), or maybe even log the whole thing.

File: test/response.validation.on.error.spec.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+  let onErrorArgs = null;
+  before(async () => {
+    // set up express app
+    app = await createApp(
+      {
+        apiSpec: apiSpecPath,
+        validateResponses: {
+          onError: function() {
+            onErrorArgs = Array.from(arguments);
```
Comment (cdimascio) [inline test/response.validation.on.error.spec.ts:20]: can you also add a test, where you re-throw the error and/or create a new error. looks like should all work out, but having a test will be good.

**Comment by cdimascio** [comment]
thanks for this @nampas

**Comment by cdimascio** [comment]
one more note, please update the validateResponses section of the README to describe this new option

**Review by cdimascio — APPROVED** [review]
@nampas thanks for the feature

**Pull request merged**